packages/manager/modules/octavia-load-balancer/src/load-balancers/load-balancer/listeners/create/controller.js

```diff
@@ -0,0 +1,94 @@
+import { PROTOCOLS, LISTENER_POOL_PROTOCOL_COMBINATION } from './constants';
+
+export default class OctaviaLoadBalancerListenersCtrl {
+  /* @ngInject */
+  constructor(OctaviaLoadBalancerListenersService, Alerter, $translate) {
+    this.OctaviaLoadBalancerListenersService = OctaviaLoadBalancerListenersService;
+    this.Alerter = Alerter;
+    this.$translate = $translate;
+    this.PROTOCOLS = PROTOCOLS;
+    this.loading = false;
+  }
+
+  $onInit() {
+    this.model = {
+      loadbalancerId: this.loadbalancerId,
+      name: '',
+      port: null,
+      protocol: null,
+      pool: null,
+    };
+    this.filteredPools = this.filterPools();
+  }
+
+  onProtocolChange() {
+    if (this.model.protocol === 'http') {
+      this.model.port = 80;
+    } else if (this.model.protocol === 'https') {
+      this.model.port = 443;
+    }
```

Comment:
You should manage it with constants
```suggestion
        this.model.port = PROTOCOL_PORT[this.model.protocol];
```
No need to default value ?